enhancements/update/change-management-and-maintenance-schedules.md

```diff
@@ -0,0 +1,1220 @@
+- Standalone's `ClusterVersion`. Honored by Cluster Version Operator.
+- Standalone's `MachineConfigPool`. Honored by Machine Config Operator.
+
+The implementation of `changeManagement` will vary by profile
+and resource, however, they will share a core schema and provide a reasonably consistent user
+experience across profiles. 
+
+The schema will provide options for controlling exactly when changes to API resources on the 
+cluster can initiate material changes to the cluster. Changes that are not allowed to be
+initiated due to a change management control will be called "pending". Subsystems responsible
+for initiating pending changes will await a permitted window according to the change's
+relevant `changeManagement` configuration(s).
+
+### Change Management Strategies
+Each resource supporting change management will add the `changeManagement` stanza and support a minimal set of change management strategies.
+Each strategy may require an additional configuration element within the stanza. For example:
+```yaml
+spec:
+  changeManagement:
+    strategy: "MaintenanceSchedule"
+    pausedUntil: false
+    disabledUntil: false
+    config:
+       maintenanceSchedule:
+         ..options to configure a detailed policy for the maintenance schedule..
+```
+
+All change management implementations must support `Disabled` and `MaintenanceSchedule`. Abstracting 
+change management into strategies allows for simplified future expansion or deprecation of strategies. 
+Tactically, `strategy: Disabled` provides a convenient syntax for bypassing any configured 
+change management policy without permanently deleting its configuration.
+
+For example, if SRE needs to apply emergency corrective action on a cluster with a `MaintenanceSchedule` change
+management strategy configured, they can simply set `strategy: Disabled` without having to delete the existing
+`maintenanceSchedule` stanza which configures the previous strategy. Once the correct action has been completed,
+SRE simply restores `strategy: MaintenanceSchedule` and the previous configuration begins to be enforced.
+
+Configurations for multiple management strategies can be recorded in the `config` stanza, but
+only one strategy can be active at a given time.
+
+Each strategy will support a policy for pausing or unpausing (permitting) material changes from being initiated. 
+This will be referred to as the strategy's enforcement state (or just "state"). The enforcement state for a
+strategy can be either "paused" or "unpaused" (a.k.a. "permissive"). The `Disabled` strategy enforcement state 
+is always permissive -- allowing material changes to be initiated (see [Change Management
+Hierarchy](#change-management-hierarchy) for caveats).
+
+All change management strategies, except `Disabled`, are subject to the following `changeManagement` fields:
+- `changeManagement.disabledUntil: <bool|date>`: When `disabledUntil: true` or `disabledUntil: <future-date>`, the interpreted strategy for 

```

Comment:
Linked above as well, but inspired by HyperShift to provide a consistent feel with their existing API: https://hypershift-docs.netlify.app/how-to/pause-reconciliation/ . That said, if it is a shaky foundation, maybe we suggest a change in HyperShift.

Reply:
Lets consolidate into a single thread. But, in the past we have tried not to "repeat sins of the past" and recommend better structures that are more easily validateable and easier for users to be explicit about their expression of wish. The bool|date field is not easy to validate in CRD schemas, and it also is difficult to unmarshal without errors (since it is implemented as a string type, so there are definitely things we could do to make it more Kube native